staging/src/k8s.io/apiserver/pkg/util/flowcontrol/match_test.go

```diff
@@ -120,162 +120,236 @@ func TestLiterals(t *testing.T) {
+			APIGroups:     []string{"goodapig"},
+			Resources:     []string{"goodrscs"},
+			Namespaces:    []string{"goodns"},
+			ResourceNames: []string{"eman"},
+		}}}, {
 		Subjects: []fcv1a1.Subject{{Kind: fcv1a1.SubjectKindGroup,
 			Group: &fcv1a1.GroupSubject{"*"}}},
 		ResourceRules: []fcv1a1.ResourcePolicyRule{{
-			Verbs:      []string{"goodverb"},
-			APIGroups:  []string{"goodapig"},
-			Resources:  []string{"goodrscs"},
-			Namespaces: []string{"goodns"}}}}, {
+			Verbs:         []string{"goodverb"},
+			APIGroups:     []string{"goodapig"},
+			Resources:     []string{"goodrscs"},
+			Namespaces:    []string{"goodns"},
+			ResourceNames: []string{"eman"},
+		}}}, {
 		Subjects: []fcv1a1.Subject{{Kind: fcv1a1.SubjectKindUser,
 			User: &fcv1a1.UserSubject{"goodu"}}},
 		ResourceRules: []fcv1a1.ResourcePolicyRule{{
-			Verbs:      []string{"*"},
-			APIGroups:  []string{"goodapig"},
-			Resources:  []string{"goodrscs"},
-			Namespaces: []string{"goodns"}}}}, {
+			Verbs:         []string{"*"},
+			APIGroups:     []string{"goodapig"},
+			Resources:     []string{"goodrscs"},
+			Namespaces:    []string{"goodns"},
+			ResourceNames: []string{"eman"},
+		}}}, {
 		Subjects: []fcv1a1.Subject{{Kind: fcv1a1.SubjectKindUser,
 			User: &fcv1a1.UserSubject{"goodu"}}},
 		ResourceRules: []fcv1a1.ResourcePolicyRule{{
-			Verbs:      []string{"goodverb"},
-			APIGroups:  []string{"*"},
-			Resources:  []string{"goodrscs"},
-			Namespaces: []string{"goodns"}}}}, {
+			Verbs:         []string{"goodverb"},
+			APIGroups:     []string{"*"},
+			Resources:     []string{"goodrscs"},
+			Namespaces:    []string{"goodns"},
+			ResourceNames: []string{"eman"},
+		}}}, {
 		Subjects: []fcv1a1.Subject{{Kind: fcv1a1.SubjectKindUser,
 			User: &fcv1a1.UserSubject{"goodu"}}},
 		ResourceRules: []fcv1a1.ResourcePolicyRule{{
-			Verbs:      []string{"goodverb"},
-			APIGroups:  []string{"goodapig"},
-			Resources:  []string{"*"},
-			Namespaces: []string{"goodns"}}}}, {
+			Verbs:         []string{"goodverb"},
+			APIGroups:     []string{"goodapig"},
+			Resources:     []string{"*"},
+			Namespaces:    []string{"goodns"},
+			ResourceNames: []string{"eman"},
+		}}}, {
 		Subjects: []fcv1a1.Subject{{Kind: fcv1a1.SubjectKindUser,
 			User: &fcv1a1.UserSubject{"goodu"}}},
 		ResourceRules: []fcv1a1.ResourcePolicyRule{{
-			Verbs:      []string{"goodverb"},
-			APIGroups:  []string{"goodapig"},
-			Resources:  []string{"goodrscs"},
-			Namespaces: []string{"*"}}}},
+			Verbs:         []string{"goodverb"},
+			APIGroups:     []string{"goodapig"},
+			Resources:     []string{"goodrscs"},
+			Namespaces:    []string{"*"},
+			ResourceNames: []string{"eman"},
+		}}}, {
+		Subjects: []fcv1a1.Subject{{Kind: fcv1a1.SubjectKindUser,
+			User: &fcv1a1.UserSubject{"goodu"}}},
+		ResourceRules: []fcv1a1.ResourcePolicyRule{{
+			Verbs:         []string{"goodverb"},
+			APIGroups:     []string{"goodapig"},
+			Resources:     []string{"goodrscs"},
+			Namespaces:    []string{"*"},
```

Comment:
I would make this `"goodns"` instead of `"*"` because this series of test cases is testing individual wildcards.

Reply:
yes, updated